tidy3d/plugins/near2far/near2far.py

```diff
@@ -1,130 +1,483 @@
+    )
+
+    @property
+    def origin(self):
+        """Average of centers of all surface monitors to use as a local origin."""
+        centers = np.array([surface.monitor.center for surface in self.surfaces])
+        return tuple(np.mean(centers, axis=0))
+
+    @pydantic.validator("medium", always=True)
+    def set_medium(cls, val, values):
+        """Sets the .medium field using the simulation default if no medium was provided."""
+        if val is None:
+            val = values.get("sim_data").simulation.medium
+        return val
+
+    @property
+    def nk(self) -> [float, float]:
+        """Returns the real and imaginary parts of the background medium's refractive index."""
+        eps_complex = self.medium.eps_model(self.frequency)
+        return self.medium.eps_complex_to_nk(eps_complex)
+
+    @property
+    def k(self) -> complex:
+        """Returns the complex wave number associated with the background medium."""
+        n, k = self.nk
+        return (2 * np.pi * self.frequency / C_0) * (n + 1j * k)
+
+    @property
+    def eta(self) -> complex:
+        """Returns the complex wave impedance associated with the background medium."""
+        n, k = self.nk
+        return ETA_0 / (n + 1j * k)
+
+    @classmethod
+    # pylint:disable=too-many-arguments
+    def from_surface_monitors(cls,
+        sim_data: SimulationData,
+        monitors: List[FieldMonitor],
+        normal_dirs: List[Direction],
+        frequency: float,
+        pts_per_wavelength: int = 10,
+        medium: Medium = None):
+        """Constructs :class:`Near2Far` from a list of surface monitors and their directions.
 
         Parameters
         ----------
         sim_data : :class:`.SimulationData`
-            Container for simulation data containing a near field monitor.
-        mon_name : str
-            Name of the :class:`.FieldMonitor` to use as source of near field.
-            Must be a :class:`.FieldMonitor` and stored in ``sim_data``.
+            Container for simulation data containing the near field monitors.
+        monitors : List[:class:`.FieldMonitor`]
+            List of :class:`.FieldMonitor` objects on which near fields will be sampled.
+        normal_dirs : List[:class:`.Direction`]
+            List containing the :class:`.Direction` of the normal to each surface monitor 
+            w.r.t. to the positive x, y or z unit vectors. Must have the same length as monitors.
         frequency : float
-            Frequency to select from the :class:`.FieldMonitor` to use for projection.
-            Must be a frequency stored in the :class:`FieldMonitor`.
+            Frequency to select from each :class:`.FieldMonitor` to use for projection.
+            Must be a frequency stored in each :class:`FieldMonitor`.
+        pts_per_wavelength : int
+            Number of points per wavelength with which to discretize the
+            surface monitors for the projection. Default: 10.
```

Comment:
instead of `Default: 10` just put `int = 10` above.  That's how I've been doing it in other parts of the code.
Also, let's define the `PTS_PER_WVL=10` constant at the beginning of this file and insert it instead of `10` where `10` appears, just to have it propagate everywhere If we change things.